File: example/javaexample/src/main/java/com/github/app/javaexample/MainActivity.java
package com.github.app.javaexample;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.github.app.javaexample.ui.IntroAdapter;
import com.github.app.javaexample.ui.IntroEntry;
import com.github.app.javaexample.ui.custom.CustomBackgroundIntro;
import com.github.app.javaexample.ui.custom.CustomLayoutIntro;
import com.github.app.javaexample.ui.custom.SlidePolicyIntro;
import com.github.app.javaexample.ui.defaullt.DefaultIntro;
import com.github.app.javaexample.ui.defaullt.DefaultIntro2;
import com.github.app.javaexample.ui.permissions.PermissionsIntro;
import com.github.app.javaexample.ui.permissions.PermissionsIntro2;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    List<IntroEntry> introEntryList;

    private RecyclerView mainRecyclerview;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainRecyclerview = findViewById(R.id.main_recycler_view);

        introEntryList = new ArrayList<>();

        mainRecyclerview.setHasFixedSize(true);
        mainRecyclerview.setLayoutManager(new LinearLayoutManager(this));
        initIntroEntry();

        IntroAdapter adapter = new IntroAdapter(introEntryList,this);
        mainRecyclerview.setAdapter(adapter);
    }

    private void initIntroEntry() {
        introEntryList.add(new IntroEntry(R.string.default_intro_title, R.string.default_intro, DefaultIntro.class));
        introEntryList.add(new IntroEntry(R.string.default_intro2_title, R.string.default_intro2, DefaultIntro2.class));
        introEntryList.add(new IntroEntry(R.string.custom_layout_intro_title, R.string.custom_layout_intro, CustomLayoutIntro.class));
        introEntryList.add(new IntroEntry(R.string.custom_background_intro_title, R.string.custom_background_intro, CustomBackgroundIntro.class));
        introEntryList.add(new IntroEntry(R.string.slide_policy_intro_title, R.string.slide_policy_intro, SlidePolicyIntro.class));
        introEntryList.add(new IntroEntry(R.string.perms_intro1_title, R.string.perms_intro1, PermissionsIntro.class));
        introEntryList.add(new IntroEntry(R.string.perms_intro2_title, R.string.perms_intro2, PermissionsIntro2.class));
    }
}

File: example/src/main/java/com/github/appintro/example/ui/MainActivity.kt
package com.github.appintro.example.ui

import android.os.Bundle
import android.os.StrictMode
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.github.appintro.example.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Enable StrictMode
        StrictMode.setThreadPolicy(
            StrictMode.ThreadPolicy.Builder()
                .detectAll()
                .penaltyLog()
                .build()
        )
        setContentView(R.layout.activity_main)

       // val viewManager = LinearLayoutManager(this)
      //  val viewAdapter = IntroAdapter(defaultEntries)

        //  val recyclerView = findViewById<RecyclerView>(R.id.main_recycler_view).apply {
          //  recyclerView.setHasFixedSize(true)
          //  layoutManager = viewManager
           // adapter = viewAdapter

           setContentView(R.layout.activity_main)
           val viewManager = LinearLayoutManager(this)
           val viewAdapter = IntroAdapter(defaultEntries)

           val recyclerView = findViewById<RecyclerView>(R.id.main_recycler_view)
           recyclerView.setHasFixedSize(true)
           recyclerView.layoutManager = viewManager
           recyclerView.adapter = viewAdapter

       }
    }
